src/IntentBuilder.ts

```diff
@@ -80,15 +85,20 @@ export class IntentBuilder {
     const intents = new Intent({
       from: this.setFrom(from),
       to: this.setTo(to),
+      recipient,
     });
 
     const calldata = ethers.toUtf8Bytes(JSON.stringify(intents));
 
-    if (destChainId && destChainId !== sourceChainId) {
-      return this.executeCrossChain(sourceChainId, destChainId, account, calldata);
-    } else {
-      return this.executeSingleChain(sourceChainId, account, calldata);
-    }
+    // TODO:: PR https://github.com/blndgs/intents.js/pull/150 should address cross chain
+    // important checks
+    // 1. aggregated userOps is delegate cross chain call and should call executeCrossChain
+    // 2. when both intent states are `Asset` and src and dest chains are diffrent that is non-delegate cross chain call and should call executeSingleChain
+    // if (destChainId && destChainId !== sourceChainId) {
+    //   return this.executeCrossChain(sourceChainId, destChainId, account, calldata);
+    // }
+
+    return this.executeSingleChain(sourceChainId, account, calldata);
```

Comment:
Not running post-hook 4337 userOps doesn't affect unless you need the msg.sender to point to the user wallet address.